lua/octo/workflow_runs.lua

```diff
@@ -0,0 +1,730 @@
+        local json = vim.fn.json_decode(table.concat(data))
+        for _, value in ipairs(json) do
+          local wf_run = {
+            status = value.status == "queued" and icons.pending
+                or value.status == "in_progress" and icons.in_progress
+                or value.conclusion == "failure" and icons.failed
+                or icons.succeeded,
+            title = value.displayTitle,
+            display = value.displayTitle,
+            value = value.databaseId,
+            branch = value.headBranch,
+            name = value.name,
+            age = utils.format_date(value.updatedAt),
+            id = value.databaseId,
+          }
+          table.insert(lines, wf_run)
+        end
+      end,
+      on_exit = function()
+        coroutine.resume(co)
+      end,
+    }
+  )
+  coroutine.yield()
+  return lines
+end
+
+local preview_picker = function(bufnr, options, on_select_cb, title, previewer)
+  if #options == 0 then
+    error "No options provided, minimum 1 is required"
+  end
+
+  -- Auto pick if only one option present
+  if #options == 1 then
+    on_select_cb(options[1])
+    return
+  end
+
+  local previewers = require "telescope.previewers"
+  local picker = require("telescope.pickers").new(bufnr, {
+    prompt_title = title,
+    finder = require("telescope.finders").new_table {
+      results = options,
+      entry_maker = function(entry)
+        return {
+          display = entry.display,
+          value = entry,
+          ordinal = entry.display,
+        }
+      end,
+    },
+    previewer = previewers.new_buffer_previewer {
+      title = title .. " preview",
+      define_preview = previewer,
+    },
+    sorter = require("telescope.config").values.generic_sorter {},
+    preview = true,
+    attach_mappings = function(_, map)
+      map("i", "<CR>", function(prompt_bufnr)
+        local selection = require("telescope.actions.state").get_selected_entry()
+        require("telescope.actions").close(prompt_bufnr)
+        on_select_cb(selection.value)
+      end)
+      map("n", "<CR>", function(prompt_bufnr)
+        local selection = require("telescope.actions.state").get_selected_entry()
+        require("telescope.actions").close(prompt_bufnr)
+        on_select_cb(selection.value)
+      end)
+      map("n", "q", function(prompt_bufnr)
+        require("telescope.actions").close(prompt_bufnr)
+      end)
+      return true
+    end,
+  })
+  picker:find()
+end
+
+local function render(selected)
+  local new_buf = vim.api.nvim_create_buf(true, true)
+  M.buf = new_buf
+  vim.api.nvim_set_current_buf(new_buf)
+  populate_preview_buffer(selected.id, new_buf)
+  vim.api.nvim_buf_set_name(new_buf, "" .. selected.id)
+end
+
+
+M.list = function()
+  vim.notify "Fetching workflow runs (this may take a while) ..."
+  local co = coroutine.running()
+  local wf_runs = get_workflow_runs_sync(co)
+
+  preview_picker(
+    nil,
+    wf_runs,
+    function(selected)
+      render(selected)
+    end,
```

Comment:
Nit. just pass `render`
```suggestion
    render,
```